react/features/dial-out/components/DialOutNumbersForm.web.js

```diff
@@ -0,0 +1,355 @@
+     * @returns {ReactElement}
+     */
+    _createDropdownMenu(items, selectedCode) {
+        return (
+            <StatelessDropdownMenu
+                isOpen = { this.state.isDropdownOpen }
+                items = { [ { items } ] }
+                onItemActivated = { this._onSelect }
+                onOpenChange = { this._onOpenChange }
+                shouldFitContainer = { true }>
+                { this._createDropdownTrigger(
+                    selectedCode.dialCode,
+                    selectedCode.code) }
+            </StatelessDropdownMenu>
+        );
+    }
+
+    /**
+     * Creates a React {@code Component} with a readonly HTMLInputElement as a
+     * trigger for displaying the dropdown menu. The {@code Component} will also
+     * display the currently selected number.
+     *
+     * @param {string} dialCode -
+     * @param {string} countryCode -
+     * @private
+     * @returns {ReactElement}
+     */
+    _createDropdownTrigger(dialCode, countryCode) {
+        return (
+            <div className = 'dropdown'>
+                <CountryIcon
+                    className = 'dial-out-flag-icon'
+                    countryCode = { `${countryCode}` } />
+                <input
+                    className = 'input-control dial-out-code'
+                    readOnly = { true }
+                    type = 'text'
+                    value = { dialCode || '' } />
+                <span className = 'dropdown-trigger-icon'>
+                    { EXPAND_ICON }
+                </span>
+            </div>
+        );
+    }
+
+    /**
+     * Updates the dialInput state when the input changes.
+     *
+     * @param {Object} e - The event notifying us of the change.
+     * @private
+     * @returns {void}
+     */
+    _onInputChange(e) {
+        this.setState({
+            dialInput: e.target.value
+        }, () => {
+            this._onDialNumberChange();
+        });
+    }
+
+    /**
+     * Updates the dialNumber when changes to the dial text or code happen.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onDialNumberChange() {
+        const dialInput = this.state.dialInput;
+
+        const dialNumber
+            = this.state.selectedCode
+            ? this.state.selectedCode.content
+                + this._formatDialInput(dialInput)
+            : dialInput;
+
+        const formattedNumber = this._formatDialNumber(dialNumber);
+
+        this.props.onChange(formattedNumber);
+    }
+
+    /**
+     * Formats the dial input text.
+     *
+     * @param {string} dialInput - The text input for dial number.
+     * @private
+     * @returns {string} - The formatted phone number.
+     */
+    _formatDialInput(dialInput) {
+        let startIndex = 0;
+
+        if (dialInput.startsWith('00')) {
+            startIndex = 2;
+        } else if (dialInput.startsWith('0') || dialInput.startsWith('+')) {
+            startIndex = 1;
+        }
+
+        return dialInput.substring(startIndex);
```

Comment:
What about:
`return dialInput.substring(0, 2).replace('0','').replace('+','') + dialInput.substring(2);`

Maybe there is even simpler solution, maybe with regex?